LandSandBoat/server
# Expand Dealer Moogle Plumbing

**_I affirm:_**

## What does this pull request do?

* Adds in missing Mog Kupon Item/Key Item IDs
* Updates Mask Generation Function to apply appropriate bit shifts
* Updates Trade logic to account for scenarios where player may already have the Key Item corresponding to the Traded Item
* Updates comments to detail script behavior
* Adds Kupon IDs to xi.items
* Flags all Mog Kupon Items as non-vendorable
* Adds all currently available Mog Kupons
* TODO: Implement/Apply Rank Augments for Mog Kupon W-RMEA

## Steps to test these changes

Reviewers: This PR evolves a lot, reviewing by commit may be helpful.

1. `!additem <some kupon id> 2` e.g. `!additem MOG_KUPON_A_DBCD 2`
2. `!addkeyitem <a different kupon id>` e.g. `!addkeyitem MOG_KUPON_AW_VGR`
3. Trade the item from Step 1 to the Dealer Moogle
4. Confirm the dialog references the item name in the trade.
5. Cancel out of Transaction
6. Confirm Key Item in chat log and Player Key Items
7. Trigger Dealer Moogle, Confirm two Key Items are presented.
8. Navigate, through either, back out of transaction dialog.
9. Trade Dealer Moogle the second Item from Step 1
10. Confirm dialog does not reference the traded item
11. Back out of transaction
12. Confirm Item from Step 1 still in inventory, and corresponding Key Item still present.
